### PR TITLE
Prevent crash on compute MachinePool with no platform.

### PR DESCRIPTION
**Pull request by dgoodwin**
Implemented via validation, as well as a safety check in the
remotemachineset_controller at the point we were hitting the nil
pointer.

### CODE REVIEW

File: pkg/apis/hive/v1alpha1/validating-webhooks/clusterdeployment_validating_admission_hook.go [hunk 1]
```diff
@@ -285,6 +297,15 @@ func (a *ClusterDeploymentValidatingAdmissionHook) validateCreate(admissionSpec
 		if gcp.Region == "" {
 			allErrs = append(allErrs, field.Required(gcpPath.Child("region"), "must specify GCP region"))
 		}
+		// TODO: Uncomment once ready for GCP
```
Comment (staebler) [inline pkg/apis/hive/v1alpha1/validating-webhooks/clusterdeployment_validating_admission_hook.go]: Should uncomment now. We don't need to wait for GCP for this validation to be here.

File: pkg/apis/hive/v1alpha1/validating-webhooks/clusterdeployment_validating_admission_hook.go [hunk 2]
```diff
@@ -257,6 +257,12 @@ func (a *ClusterDeploymentValidatingAdmissionHook) validateCreate(admissionSpec
 		if aws.Region == "" {
 			allErrs = append(allErrs, field.Required(awsPath.Child("region"), "must specify AWS region"))
 		}
+		for i, mp := range newObject.Spec.Compute {
+			computePath := specPath.Child("compute").Index(i)
+			if mp.Platform.AWS == nil {
```
Comment (staebler) [inline pkg/apis/hive/v1alpha1/validating-webhooks/clusterdeployment_validating_admission_hook.go]: Why is this required? The user should be able to omit it and use the defaults.

File: pkg/apis/hive/v1alpha1/validating-webhooks/clusterdeployment_validating_admission_hook.go (hunk 1 above)
Comment (staebler) [reply pkg/apis/hive/v1alpha1/validating-webhooks/clusterdeployment_validating_admission_hook.go]: Oh, duh. The fields are not in the machine pool platform, yet.

File: pkg/apis/hive/v1alpha1/validating-webhooks/clusterdeployment_validating_admission_hook.go (hunk 2 above)
Comment (dgoodwin) [reply pkg/apis/hive/v1alpha1/validating-webhooks/clusterdeployment_validating_admission_hook.go]: Looking into this now, kinda looks like default machine pool platform may not be used.

Comment (staebler) [reply pkg/apis/hive/v1alpha1/validating-webhooks/clusterdeployment_validating_admission_hook.go]: The default machine pool is something that is applied by the installer. The order of precedence for the value of a field in the machine pool platform is as follows.
1) The value set in the machine pool platform for a specific machine pool.
2) The value set in the default machine pool platform for the cluster.
3) The default value for the field.

Comment (staebler) [reply pkg/apis/hive/v1alpha1/validating-webhooks/clusterdeployment_validating_admission_hook.go]: My concern is that the remotemachineset controller is not using the same algorithm for determining what values it expects on the resultant machine set.
